### DISCUSSION TITLE
Intel i7 1-CPU, blitz-time, hash size "sweet spot" rule of thumb?

### DISCUSSION THREAD

**Discussion by yuribudilov** [official-stockfish/Stockfish#3826] [Q&A] [answered]
Hello everyone

On my home laptop running Stockfish 14.1 engine if I am using a **very short time control** and 1 logical CPU core, ponder-is-OFF.
Consider TC of 1-3 minutes plus +5 seconds per move time increment on 1-CPU 1-logical core, Intel i7 processor. 
My question is: what is the good rule-of-thumb optimal hash size to use for this? 
I do realize 16MB "default" is very small on modern CPUs.

I noticed in (online) computer testing they use 1 CPU with blitz time controls with 1-5 seconds per move time increment too and their hash size is only =256 MB which on modern hardware is very small. 

I have run some very rudimentary testing of hash 256MB versus 2GB hash and I did not see any significant strength improvements using 1 CPU core, no-ponder, on same ELO level.

So my question is - is there a sweet-spot of hash tables memory size you could recommend?

thank you very much

**Comment by vondele** [comment, marked as answer]
the rule of thumb I use is to run a search from the starting position, and look at the Hashfull output of the engine with time/depth. For the time that is typical of your time control, the hash should ideally be around 50-70% full.

Having said that, the impact of hash sizes that are different by even a 4x factor will be a few Elo only, hard to measure unless you play 1000s of games.

**Reply by yuribudilov** [reply]
thank you very much! My very unscientific/rudimentary testing shows that also, on a 1 CPU thread there is not much (or even any?) strength difference on rapid/blitz controls. For longer game analysis (30-60 sec per move) I do use larger hash, 2-4 GB for 1 CPU-core.
